tests/end-to-end/external-solution-integration/pkg/testkit/compass_director.go

```diff
@@ -0,0 +1,235 @@
+package testkit
+
+import (
+	"context"
+	"github.com/avast/retry-go"
+	"github.com/kyma-incubator/compass/components/director/pkg/graphql"
+	"github.com/kyma-incubator/compass/tests/end-to-end/pkg/gql"
+	"github.com/kyma-incubator/compass/tests/end-to-end/pkg/idtokenprovider"
+	"github.com/kyma-project/kyma/tests/end-to-end/external-solution-integration/pkg/helpers"
+	gcli "github.com/machinebox/graphql"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"k8s.io/client-go/kubernetes"
+	"os"
+	"reflect"
+	"strings"
+	"time"
+)
+
+type CompassDirectorClient struct {
+	gqlClient *gcli.Client
+	graphqlizer gql.Graphqlizer
+	fixtures helpers.CompassFixtures
+	state CompassDirectorClientState
+}
+
+type CompassDirectorClientState interface {
+	GetScenariosLabelKey() string
+	GetDefaultTenant() string
+	GetRuntimeID() string
+	GetDexSecret() (string, string)
+}
+
+func NewCompassDirectorClient(coreClient *kubernetes.Clientset, state CompassDirectorClientState) (*CompassDirectorClient, error) {
```

Comment:
This client could be also moved to compass repo, especially if we use it in our tests